scanner/mappers/mappers.go

```diff
@@ -274,20 +274,20 @@ func toProtoV4Environment(e *claircore.Environment) *v4.Environment {
 	}
 }
 
-func toProtoV4PackageVulnerabilitiesMap(ccPkgVulnerabilities map[string][]string) map[string]*v4.StringList {
+func toProtoV4PackageVulnerabilitiesMap(ccPkgVulnerabilities map[string][]string, ccVulnerabilities map[string]*claircore.Vulnerability) map[string]*v4.StringList {
 	if ccPkgVulnerabilities == nil {
 		return nil
 	}
 	var pkgVulns map[string]*v4.StringList
 	if len(ccPkgVulnerabilities) > 0 {
 		pkgVulns = make(map[string]*v4.StringList, len(ccPkgVulnerabilities))
 	}
-	for k, v := range ccPkgVulnerabilities {
-		if v == nil {
+	for id, vulnIDs := range ccPkgVulnerabilities {
+		if vulnIDs == nil {
 			continue
 		}
-		pkgVulns[k] = &v4.StringList{
-			Values: append([]string(nil), v...),
+		pkgVulns[id] = &v4.StringList{
+			Values: filterVulns(vulnIDs, ccVulnerabilities),
```

Comment:
Nitpick: You can probably improve this by creating one map of `{<id>: <actual id>}` outside this loop. Then, here, you map vulnsIDs to their actual IDs, removing dups. Maybe worth it to avoid creating the `vulnsByName` map for each package.

Reply:
are you thinking of a global filter for vulnerabilities? I think doing it by package is best, as it could be possible two packages are affected by the same CVE but with differing severities. Let me know if I'm misunderstanding